### ISSUE TITLE
关于nni问题#2000 default metric graphs are empty after completing run of examples\trials\mnist-pytorch\config_windows.yml 的问题

### ISSUE THREAD

**Issue by 99hansling** [microsoft/nni#2399]
## NNI 学生项目问题概述 / General Question of Student Program

**请简要概述您的问题 / 观点 :**
**Short summary about the question / idea :**

我也出现了类似的问题：就是使用pytorch的gpu模式跑trial的时候，default metrix不能显示
在最后，我下载了naive_example.zip 并进行了测试，这里我将提供结果,供参考，以便最终解决metrix的显示问题

使用naive test的话是可以显示default metrix的，但是trial是不可以的。希望您能告知原因。非常感谢！！

**请提供 NNI 环境信息 :**
**nni Environment :**
- nni version:1.5
- nni mode(local|pai|remote):local
- OS:windows 10
- python version:3.7
- is conda or virtualenv used?:no
- is running in docker?:no
[naive_example.zip](https://github.com/microsoft/nni/files/4559357/naive_example.zip)
[输出结果.zip](https://github.com/microsoft/nni/files/4559695/default.zip)

## 其他建议 / Other Advice

**是否需要更新文档（是 / 否）:**
**Need to update document ( yes / no ) :**

**其他分享内容 :**
**Anything else we need to know :**

**Log message / 日志信息 :**
 - [nnimanager.log and dispatcher.log](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/HowToDebug.md#experiment-root-directory) : 

 - [nnictl stdout and stderr](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/Nnictl.md#nnictl%20log%20stdout) :

**Comment by ultmaster** [comment]
说明你 pytorch 的 trial 根本没跑成功。我看了一下你的 trial 目录下的 stdout.log 和 stderr 都是空的。你可以试下单独跑那个 pytorch 的程序会怎么样。

**Comment by 99hansling** [comment]
> 说明你 pytorch 的 trial 根本没跑成功。我看了一下你的 trial 目录下的 stdout.log 和 stderr 都是空的。你可以试下单独跑那个 pytorch 的程序会怎么样。
抱歉让您误会，但是这里的输出文件实际上是运行naive_example得到的输出。
这里我贴上pytorch-gpu跑MNIST的结果
[MNIST运行结果.zip](https://github.com/microsoft/nni/files/4580213/MNIST.zip)

**Comment by ultmaster** [comment]
你可以看一下 trials\xxxxx\stderr 和 trial.log 下的输出，和单独运行 pytorch 的程序输出是否一致，运行是否成功。如果能正常运行的话，试下在 pytorch 的代码里多加几个 nni.report_intermediate_result 或者 nni.report_final_result 看看不能正常收到。

**Comment by 99hansling** [comment]
> 你可以看一下 trials\xxxxx\stderr 和 trial.log 下的输出，和单独运行 pytorch 的程序输出是否一致，运行是否成功。如果能正常运行的话，试下在 pytorch 的代码里多加几个 nni.report_intermediate_result 或者 nni.report_final_result 看看不能正常收到。

1.我尝试了增加nni.report_intermediate_result和 nni.report_final_result，不知道是否是这样
![20200506113654](https://user-images.githubusercontent.com/44934784/81137034-e1d2b800-8f8f-11ea-90a9-06a52eccd1b8.png)
2.然后这里是直接MNIST.py时候在命令行中的输出：
![20200506114649](https://user-images.githubusercontent.com/44934784/81137054-f7e07880-8f8f-11ea-9fa4-0d5a07434b79.png)
3.我觉得这个和stderr差不多，下面是用nnictl执行的一个stderr（增加nni.report之后）：
[stderr.zip](https://github.com/microsoft/nni/files/4584747/stderr.zip)
4.然后这里是trial.log（无论是否加nni.report前还是后都是这样），可以看出明显和正常运行的不一样
![20200506114801](https://user-images.githubusercontent.com/44934784/81137243-b7cdc580-8f90-11ea-854a-5063c1239d61.png)
[trial.log](https://github.com/microsoft/nni/files/4584749/trial.log)

5.然后是metric：
这里是加入nni.report 前：
![20200506114900](https://user-images.githubusercontent.com/44934784/81137300-f06d9f00-8f90-11ea-93a5-55af9e3b8c7d.png)
[metrics.zip](https://github.com/microsoft/nni/files/4584756/metrics.zip)


这是加入nni.report后：
![20200506114830](https://user-images.githubusercontent.com/44934784/81137364-227f0100-8f91-11ea-8f34-264b1b8246dd.png)

[metrics.zip](https://github.com/microsoft/nni/files/4584752/metrics.zip)

但是最终还是显示no data：
![20200506115019](https://user-images.githubusercontent.com/44934784/81137406-4b06fb00-8f91-11ea-8f23-a2bc207a138d.png)

![20200506120147](https://user-images.githubusercontent.com/44934784/81137431-66720600-8f91-11ea-8b5e-b8ade73eb5d3.png)

**Comment by ultmaster** [comment]
这个问题和 #2000 是不是一样的？

遗憾的是，我们目前并没有对这个问题的解法。对你来说最方便的方案是更换一个环境，比如用原生 Linux 或者 WSL。

**Comment by SparkSnail** [comment]
Hi @99hansling, does tensorflow or keras examples works in your environment?

**Comment by 99hansling** [comment]
> Hi @99hansling, does tensorflow or keras examples works in your environment?

I did install tensorflow-gpu in my python site-package，but I think I didn‘t run tensorflow this time because tensorflow were not included in my python source code . As for keras，I’ve never installed it.

**Comment by SparkSnail** [comment]
Hi @99hansling, I mean that did you run examples like `mnist-tfv1`, `mnist-keras`, `sklearn` etc, did these example can get default metrics? Is this issue only for `pytorch` example? If you do not use gpu when running this pytorch example, does the example has metrics?

**Comment by 99hansling** [comment]
> Hi @99hansling, I mean that did you run examples like `mnist-tfv1`, `mnist-keras`, `sklearn` etc, did these example can get default metrics? Is this issue only for `pytorch` example? If you do not use gpu when running this pytorch example, does the example has metrics?

1. I‘ve run mnist-tf2（cpu mode), and it can get metrics.
2. In my experience，the issue only for “pytorch-gpu”（specifically running on windows10 in local mode）.And I’ve run pytorch in cpu mode， which can get metrics

**Comment by SparkSnail** [comment]
Hi @99hansling, sorry for late reaponse. are you using cifar10 example? If you are using cifar10 example, there is a `progress_bar` function on this example, I guess this function might write some unknow character to files,  https://github.com/microsoft/nni/blob/master/examples/trials/cifar10_pytorch/main.py#L18. Or other pytorch examples could have their own logger function, could you please annotate these log functions, and try again?

**Comment by 99hansling** [comment]
> Hi @99hansling, sorry for late reaponse. are you using cifar10 example? If you are using cifar10 example, there is a `progress_bar` function on this example, I guess this function might write some unknow character to files, https://github.com/microsoft/nni/blob/master/examples/trials/cifar10_pytorch/main.py#L18. Or other pytorch examples could have their own logger function, could you please annotate these log functions, and try again?

Actually，i’ve only used it in MNIST example and found this problem。 You mean I can annotate the all the “logger”s in this example and try again？

**Comment by SparkSnail** [comment]
> > Hi @99hansling, sorry for late reaponse. are you using cifar10 example? If you are using cifar10 example, there is a `progress_bar` function on this example, I guess this function might write some unknow character to files, https://github.com/microsoft/nni/blob/master/examples/trials/cifar10_pytorch/main.py#L18. Or other pytorch examples could have their own logger function, could you please annotate these log functions, and try again?
> 
> Actually，i’ve only used it in MNIST example and found this problem。 You mean I can annotate the all the “logger”s in this example and try again？

Yes, we could detect if the log content have any influence on NNI's metric content in this way.

**Comment by 99hansling** [comment]
> > > Hi @99hansling, sorry for late reaponse. are you using cifar10 example? If you are using cifar10 example, there is a `progress_bar` function on this example, I guess this function might write some unknow character to files, https://github.com/microsoft/nni/blob/master/examples/trials/cifar10_pytorch/main.py#L18. Or other pytorch examples could have their own logger function, could you please annotate these log functions, and try again?
> > 
> > 
> > Actually，i’ve only used it in MNIST example and found this problem。 You mean I can annotate the all the “logger”s in this example and try again？
> 
> Yes, we could detect if the log content have any influence on NNI's metric content in this way.


[mnist.py.zip](https://github.com/microsoft/nni/files/4632932/mnist.py.zip)
well,I tried to annotate the ```logger```s in this .py source code. BUT I found out that if I annotate the ```logger```s in either the function of "train" or ```test```, the trial would fail immediately, and if I annotate other ```logger```s without annotating the loggers in "train" and ```test```, the trial can run and succeed, but metrics still can't manifest anything except "no data" in web UI.

**Comment by SparkSnail** [comment]
[mnist.zip](https://github.com/microsoft/nni/files/4633137/mnist.zip)
try this code.

**Comment by 99hansling** [comment]
> [mnist.zip](https://github.com/microsoft/nni/files/4633137/mnist.zip)
> try this code. @99hansling

well. it just doesn't work. Maybe it's because my OS is windows10.

**Comment by kvartet** [comment]
@99hansling I’m closing this issue as it has no updates from user for 3 months, please feel free to reopen if you are still seeing it an active issue.
